src/plugins/unified_doc_viewer/public/components/doc_viewer_table/table.tsx

```diff
@@ -454,3 +458,38 @@ export const DocViewerTable = ({
     </EuiFlexGroup>
   );
 };
+
+function useElementWidth(className: string) {
```

Comment:
This is definitely similar to what I'm thinking, although in an actual implementation we could probably use the EUI [`useResizeObserver`](https://eui.elastic.co/#/utilities/resize-observer#useresizeobserver-hook) hook and put a `ref` on one of the table parents to track the size.